app/Services/VersionService.php

```diff
@@ -90,15 +75,68 @@ public function publishNewVersionFromGithub(string $tag)
             throw new VersionAlreadyExistsException();
         }
 
+        try {
+            $normalisedVersion = $this->versionParser->normalize($tag);
+        } catch (UnexpectedValueException) {
+            Log::error(sprintf('Invalid release channel %s', $tag));
+            throw new ReleaseChannelNotFoundException();
+        }
+
+        $releaseChannel = PluginReleaseChannel::where('name', VersionParser::parseStability($normalisedVersion))
+            ->first();
+
+        if (!$releaseChannel) {
+            Log::error(sprintf('Invalid release channel %s', $tag));
+            throw new ReleaseChannelNotFoundException();
+        }
+
         // Create the version
         $newVersion = Version::create(
             [
-                'version' => $tag
+                'version' => $tag,
+                'plugin_release_channel_id' => $releaseChannel->id,
             ]
         );
 
         // Retire old versions
-        Version::where('id', '<>', $newVersion->id)->delete();
+        Version::where('id', '<>', $newVersion->id)
+            ->get()
+            ->reject(function (Version $version) use ($newVersion) {
+                return Comparator::greaterThan(
+                    $version->version,
+                    $newVersion->version
+                );
+            })
+            ->each(function (Version $version) {
```

Comment:
Arrow function in lieu of new php version for this?

Reply:
Oooh, go on then

Reply:
You can probably do the each callback too, for extra goodness

Reply:
Not sure if I can - fn functions returns the result of expr, which could short-circuit the each if they're falsey :)